packages/dds/tree/src/feature-libraries/detachedFieldIndexSummarizer.ts

```diff
@@ -7,44 +7,90 @@ import { bufferToString } from "@fluid-internal/client-utils";
 import type { IChannelStorageService } from "@fluidframework/datastore-definitions/internal";
 import type {
 	IExperimentalIncrementalSummaryContext,
-	ISummaryTreeWithStats,
 	ITelemetryContext,
+	MinimumVersionForCollab,
 } from "@fluidframework/runtime-definitions/internal";
-import { createSingleBlobSummary } from "@fluidframework/shared-object-base/internal";
+import {
+	getConfigForMinVersionForCollab,
+	lowestMinVersionForCollab,
+	type SummaryTreeBuilder,
+} from "@fluidframework/runtime-utils/internal";
 
 import type { DetachedFieldIndex } from "../core/index.js";
-import type {
-	Summarizable,
-	SummaryElementParser,
-	SummaryElementStringifier,
+import {
+	type Summarizable,
+	type SummaryElementParser,
+	type SummaryElementStringifier,
+	VersionedSummarizer,
 } from "../shared-tree-core/index.js";
 import type { JsonCompatibleReadOnly } from "../util/index.js";
+import { FluidClientVersion } from "../codec/index.js";
 
 /**
  * The storage key for the blob in the summary containing schema data
  */
 const detachedFieldIndexBlobKey = "DetachedFieldIndexBlob";
 
 /**
- * Provides methods for summarizing and loading a tree index.
+ * The versions for the detached field index summary.
+ */
+export const enum DetachedFieldIndexSummaryVersion {
+	/**
+	 * Version 1. This version adds metadata to the SharedTree summary.
+	 */
+	v1 = 1,
+	/**
+	 * The latest version of the detached field index summary. Must be updated when a new version is added.
+	 */
+	vLatest = v1,
+}
+
+const supportedReadVersions = new Set<DetachedFieldIndexSummaryVersion>([
+	DetachedFieldIndexSummaryVersion.v1,
```

Comment:
It seems a bit confusing not to have an entry to represent the current version: `supportedReadVersions.has(minVersionToDetachedFieldIndexSummaryVersion("2.72.0"))` returns false despite the fact that such a format is supported. 

In the message codec this was handled by having `MessageFormatVersion.undefined = 0`. I'm not sure if there are better patterns, but it seems better than nothing.

Reply:
It does seem better, but why is `vPreVersioning` not included in `supportedReadVersions`? My comment about `supportedReadVersions.has(minVersionToDetachedFieldIndexSummaryVersion("2.72.0"))` returning false still holds, right?